runtime/common/RuntimeMLIR.cpp

```diff
@@ -201,4 +201,39 @@ void registerToIQMJsonTranslation() {
         return cudaq::translateToIQMJson(op, output);
       });
 }
+
+ExecutionEngine *createQIRJITEngine(ModuleOp &moduleOp) {
+  ExecutionEngineOptions opts;
+  opts.transformer = [](llvm::Module *m) { return llvm::ErrorSuccess(); };
+  opts.jitCodeGenOptLevel = llvm::CodeGenOpt::None;
+  opts.llvmModuleBuilder =
+      [&](Operation *module,
+          llvm::LLVMContext &llvmContext) -> std::unique_ptr<llvm::Module> {
+    llvmContext.setOpaquePointers(false);
+
+    auto *context = module->getContext();
+    PassManager pm(context);
+    std::string errMsg;
+    llvm::raw_string_ostream errOs(errMsg);
+    std::string qirBasePipelineConfig =
+        "func.func(quake-add-deallocs),quake-to-qir";
```

Comment:
I wonder where should we draw the line between what is part of the target pipeline and what is not. Would it make sense to add those passes to the quantinuum target pipeline? (Thus this function gets a `ModuleOp` that is already in LLVM dialect and the only step left would be the translation to LLVMIR)